File: hexchat.wp.deadbeef.py
import subprocess

import hexchat

__module_name__ = 'DeaDBeeF'
__module_version__ = '0.0.1'
__module_description__ = 'DeaDBeef currently playing script'


def now_playing(*dummy):
    proc = subprocess.check_output(['deadbeef',
                                    '--nowplaying', '"%a - %t 「%b」 %e/%l"'])
    hexchat.command('SAY np: ' + proc.decode('utf-8').strip('"'))
    return hexchat.EAT_ALL


def unload_callback(dummy):
    hexchat.prnt('RIP deadbeef wp script')

hexchat.hook_command('wp', now_playing, help='"/wp" to display currently playing deadbeef track')
hexchat.hook_unload(unload_callback)
hexchat.prnt('hexchat.wp.deadbeef.py loaded')

File: hexchat.wp.deadbeef.lua
hexchat.register("DEADBEEF", "1", "DeaDBeeF currently playing script")

local function get_np()
    local handle = assert(io.popen('deadbeef --noplaying "%a - %t 「%b」"'))
    local res = string.gsub(assert(handle:read("*a")), "\n", "")
    handle:close()
    hexchat.command("SAY np: " .. res)
    return hexchat.EAT_ALL
end

hexchat.hook_command("wp", get_np, '"/wp" to display currently playing DeaDBeeF track')
hexchat.print("hexchat.wp.deadbeef.lua loaded")
